gateway/ln-gateway/src/lib.rs

```diff
@@ -251,15 +252,26 @@ impl Gateway {
         Ok(())
     }
 
-    async fn handle_get_info(&self, _payload: InfoPayload) -> Result<GatewayInfo> {
-        let federations = self
+    async fn handle_get_info(&self, payload: InfoPayload) -> Result<GatewayInfo> {
+        let all_federations: Vec<FederationInfo> = self
             .actors
             .lock()
             .await
             .iter()
             .map(|(_, actor)| actor.get_info().expect("Failed to get actor info"))
             .collect();
 
+        let federations = payload
+            .federation_id
+            .map(|id| {
+                all_federations
+                    .iter()
+                    .filter(|info| info.federation_id.eq(&id))
```

Comment:
Just occurred to me - can there be more federations with the same ID? If not, we should probably use `find` instead of `filter` and respond with an error if there's no match, no?

Reply:
It's a threshold public key, right? I guess it's possible if someone just took configs and updated networking and ran a second instance with the same configs ... cc @jkitman

Reply:
Yeah it's the threshold pubkey so it should be unique to the federation.  The gateway should probably validate the id before connecting to a new federation to prevent attackers from spoofing as the federation, this should be possible already using the client config.

Reply:
Thanks, guys! I guess it won't be a problem anymore as I updated it to get a selected actor by the given federation_id.

Do you think that might still be a problem?

Reply:
I think we can assume the federation id is unique.

Reply:
Great! It's safe as it's then 😁